automation-scripts/minibroker-charts/extractor.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+# Script to extract working versions of minibroker database charts from helm stable repo's index.yaml.
+# Make sure the YAML parsing lib is installed: `pip install ruamel.yaml`.
+# 
+# Usage:
+# 1. Update the versions.yaml with the minibroker DB versions known to be working with current CAP release.
+# 2. Run `python extractor.py`
+
+import os
+from ruamel.yaml import YAML
+
+os.system("curl -s https://kubernetes-charts.storage.googleapis.com/index.yaml > helm_stable_index.yaml")
+
+def read_yaml(file):
+    try:
+        with open(file, 'r') as stream:
+            yaml = YAML()
+            return yaml.load(stream)
+    except yaml.YAMLError as exc:
+        print(exc)
+
+all_versions = read_yaml("helm_stable_index.yaml")
+
+def dump_yaml(data):
+    with open('index.yaml', 'w') as outfile:
+        yaml = YAML()
+        data = {"apiVersion": all_versions["apiVersion"], "entries": data}
+        yaml.dump(data, outfile)
+
+def exists(db_working_versions, db_version):
```

Comment:
What is the needle we are looking for, what is the haystack ?
(Better argument names needed (needle, and haystack ;) ).
(`db_working_versions` seems to be the hackstack, and `db_version` the needle)

Haystack seems to be an array.
Better make it a hash (python: dictionary ?).
Enables check for existence in one step instead of iterating over the array (linear search).

Reply:
Might not even require a helper method/function anymore.